src/util/stream_generator.h

```diff
@@ -151,19 +169,18 @@ class stream_generator_iterator {
   }
 
   stream_generator_iterator& operator++() {
-    _value.clear();
-    _coroutine.promise().value().str(std::move(_value));
-    // if the coroutine is done but the remaining aggregated
-    // buffer has not been cleared yet the iterator needs to be incremented one
-    // last time
-    if (!_coroutine.done()) {
-      _coroutine.resume();
-    }
-    if (_coroutine.done()) {
-      _coroutine.promise().rethrow_if_exception();
+    // Process overflow first, true means that the buffer still has capacity
+    // left.
+    if (coroutine_.promise().commitOverflow()) {
+      if (!coroutine_.done()) {
+        coroutine_.resume();
+      }
+      if (coroutine_.done()) {
+        coroutine_.promise().rethrow_if_exception();
+      }
     }
 
-    _value = std::move(_coroutine.promise().value()).str();
+    value_ = std::string{coroutine_.promise().value()};
```

Comment:
Why does the iterator here need to materialize into a `std::string` and can't just return the string_view from the `promise_type` at `operator*`?

If someone else needs a materialized string, then they have to copy it themselves.